docs/source/usage_guides/deepspeed_multiple_model.md

```diff
@@ -0,0 +1,215 @@
+specific language governing permissions and limitations under the License.
+
+⚠️ Note that this file is in Markdown but contains specific syntax for our doc-builder (similar to MDX) that may not be
+rendered properly in your Markdown viewer.
+-->
+
+# Using multiple models with DeepSpeed
+
+<Tip warning={true}>
+
+    This guide assumes that you have read and understand the [DeepSpeed usage guide](./deepspeed.md).
+
+</Tip>
+
+## Use cases
+
+There are a few use cases where one may want to run multiple models when using `accelerate` and `DeepSpeed`:
+
+* Knowledge distillation
+* RLHF techniques (see the `trl` repo for [more examples](https://github.com/huggingface/trl))
+* Training multiple models at once.
+
+As it stands, Accelerate has a **very experimental API** to help you accomplish this.
+
+This tutorial will focus on two common use cases:
+
+1. Using `zero2` to train a model, and `zero3` to perform inference for assisting of the training for the `zero2` model (as `zero2` is faster to train on), which is shown in the RLHF example
+2. Training multiple *disjoint* models at once, shown later
+
+## RLHF (multiple models, only one is trained)
+
+Normally, you would use a single [`utils.DeepSpeedPlugin`] at a time, however in this case we have two seperate configurations. Accelerate allows you to create and use multiple plugins **if and only if** they are in a `dict` so that we can reference/enable the proper plugin when needed:
+
+```python
+from accelerate.utils import DeepSpeedPlugin
+
+zero2_plugin = DeepSpeedPlugin(hf_ds_config="zero2_config.json")
+zero3_plugin = DeepSpeedPlugin(hf_ds_config="zero3_config.json")
+
+deepspeed_plugins = {"student": zero2_plugin, "teacher": zero3_plugin}
+```
+
+When doing so, `zero2_config.json` should be configured for full training (so specifying `scheduler` and `optimizer` if not utilizing your own) while `zero3_config.json` should *only* be configured for the model,
+such as:
+
+```json
+{
+    "bf16": {
+        "enabled": "auto"
+    },
+    "zero_optimization": {
+        "stage": 3,
+        "offload_param": {
+            "device": "cpu",
+            "pin_memory": true
+        },
+        "overlap_comm": true,
+        "sub_group_size": 1e9,
+        "reduce_bucket_size": 1e9,
+        "stage3_prefetch_bucket_size": 1e9,
+        "stage3_param_persistence_threshold": 1e9,
+        "stage3_max_live_parameters": 1e9,
+        "stage3_max_reuse_distance": 1e9,
+        "stage3_gather_16bit_weights_on_model_save": true
+    },
+    "train_micro_batch_size_per_gpu": 1
+}
+```
+
+<Tip>
+
+    DeepSpeed will raise an error if we don't specify a `train_micro_batch_size_per_gpu`, despite not really 
+    training this particular model
+
+</Tip>
+
+From here, we can create a single [`Accelerator`] and pass in both configurations:
+
+```python
+from accelerate import Accelerator
+
+accelerator = Accelerator(deepspeed_plugin=deepspeed_plugins)
```

Comment:
can this be fixed to `deepspeed_plugins` as the key, otherwise this is a poor API.

Perhaps for BC it could be either?

Reply:
This is a DS bug due to sanity checking never getting updated for inference. Please share a repro and issue. Thanks!